feat: move comment mutations to dedicated api routes

FabianSanchezD opened the pull request:
# Move Comment Mutations to API Routes 

This PR closes #630.

## Description
Migrated insertComment and updateComment from client components to new API routes under /api/comments. This ensures all database mutations happen securely on the server.

## Changes

Created API routes:
POST /api/comments -> create comments
PATCH /api/comments/[id] -> update comments

Moved Supabase logic & validation into API routes with authentication checks.

Ensured no server-only logic is exposed in the client bundle.

## Security

All DB mutations now server-side only

Added request validation and proper error handling

Authentication/authorization enforced

## Acceptance Criteria

✅ API routes created under /api/comments

✅ Client no longer imports server functions

✅ Comment mutations go through API routes

✅ Validation + error handling in place

✅ Server-only logic secured

## Feedback

Any feedback/code reviews are greatly appreciated!

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Create comments (questions, answers, replies) via a new comments API (POST).
  * Update existing comments and mark questions resolved via the comments API (PATCH).

* **Improvements**
  * Improved input validation and clearer error responses for create/update flows.
  * Guest posting supported (submit without signing in).
  * Status stored as metadata.status (e.g., "resolved") for consistent state handling.

* **Refactor**
  * Comment mutations moved to server API endpoints; client now uses API-driven flows.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

FabianSanchezD committed:
feat: added comment mutations to separate api routes

FabianSanchezD committed:
fix: field error with tables

Bran18 commented on apps/web/components/shared/qa-client.tsx:
```diff
@@ -476,29 +467,40 @@
-				is_resolved: true,
-			} as TablesUpdate<'comments'>)
-			return data
+			if (!effectiveUser?.id) throw new Error('User ID is required')
+			const res = await fetch(`/api/comments/${questionId}`, {
+				method: 'PATCH',
+				headers: { 'Content-Type': 'application/json' },
+				body: JSON.stringify({ is_resolved: true }),
+			})
+			const json = await res.json()
+			if (!res.ok) throw new Error(json.error || 'Failed to mark resolved')
+			return json.data as Tables<'comments'>
```
@coderabbitai create a new issue for this one

Bran18 committed:
Merge branch 'develop' into feat-630-move-comment-mutations-to-dedicated-api-routes

Bran18 commented:
@FabianSanchezD I provided feedback on one of the CoderabbitAI suggestions for you to address. Also, make sure to double-check if the build is working by running "bun run build" from the /web directory.

FabianSanchezD committed:
fix: update comment ownership check and improve error handling in PATCH (api/comments/[id[) route

FabianSanchezD committed:
Merge branch 'feat-630-move-comment-mutations-to-dedicated-api-routes' of github.com:kindfi-org/kindfi into feat-630-move-comment-mutations-to-dedicated-api-routes

FabianSanchezD commented:
Fixed the issues! Found the following problems with the past code:

Params was a Promise, I initially didn't consider that.
Added the ownership verification on the /api/comments/[id] so that users can't update other users comments.

Thanks for the code review!

AndlerRL committed:
Merge branch 'develop' into feat-630-move-comment-mutations-to-dedicated-api-routes

Bran18 committed:
Merge branch 'develop' into feat-630-move-comment-mutations-to-dedicated-api-routes

Bran18 reviewed (approved):
LGTM. Nice job
